src/domain_decomposition.hpp

```diff
@@ -13,97 +16,102 @@
 namespace nest {
 namespace mc {
 
-// Meta data used to guide the domain_decomposition in distributing
-// and grouping cells.
-struct group_rules {
-    cell_size_type target_group_size;
-    backend_policy policy;
+inline bool has_gpu_backend(cell_kind k) {
+    if (k==cell_kind::cable1d_neuron) {
+        return true;
+    }
+    return false;
+}
+
+/// Utility type for meta data for a local cell group.
+class group_description {
+    const cell_kind kind_;
+    const std::vector<cell_gid_type> gids_;
+    const backend_kind backend_;
+
+public:
+    group_description(cell_kind k, std::vector<cell_gid_type> g, backend_kind b):
+        kind_(k), gids_(std::move(g)), backend_(b)
+    {}
+
+    cell_kind kind() const {
+        return kind_;
+    }
+
+    backend_kind backend() const {
+        return backend_;
+    }
+
+    const std::vector<cell_gid_type>& gids() const {
+        return gids_;
+    }
 };
```

Comment:
Agreed. An earlier version stored the gid information in a compact form (begin and end index pair) and generated the gid indexes on the fly when requested.
Now it is just a dumb wrapper.